libevent-sys/build.rs

```diff
@@ -1,18 +1,163 @@
+            .define("EVENT__DISABLE_REGRESS", "ON")
+            .define("EVENT__DISABLE_SAMPLES", "ON");
+
+        // TODO: Address policy warnings for cross-compilation?
+        /*config.define("CMAKE_POLICY_DEFAULT_CMP0056", "NEW")
+            .define("CMAKE_POLICY_DEFAULT_CMP0066", "NEW")*/
+    }
+
+    // TODO: Can we tap into "-vv" cargo argument?
+    if cfg!(feature = "verbose_build") {
+        config.very_verbose(true);
+    }
+
+    let dst = config.build();
+
+    println!("cargo:rustc-link-search={}/lib", dst.display());
+
+    // Library 'event' is considered deprecated, so link each sub-component
+    // individually.
+    println!("cargo:rustc-link-lib=static=event_core");
+    println!("cargo:rustc-link-lib=static=event_extra");
+
+    if cfg!(feature = "openssl") || cfg!(feature = "openssl_bundled") {
+        println!("cargo:rustc-link-lib=static=event_openssl");
+    }
+
+    if cfg!(feature = "threading") {
+        println!("cargo:rustc-link-lib=static=event_pthreads");
+    }
+
+    println!("cargo:include={}/include", dst.display());
+
+    dst
+}
+
+#[cfg(not(feature = "bundled"))]
+fn run_pkg_config() -> Option<Vec<String>> {
+    let mut pkg = pkg_config::Config::new();
+    pkg.cargo_metadata(false)
         .atleast_version("2")
-        .statik(cfg!(feature = "static"))
-        .probe("libevent")
-        .unwrap();
+        .statik(cfg!(feature = "static"));
+
+    let mut lib = match pkg.probe("libevent_core") {
+        Ok(lib) => lib,
+        Err(_e) => { return None; }
+    };
+    // TODO: Probably combine all pc includes, just to be safe.
+
+    pkg.cargo_metadata(true).probe("libevent_core").unwrap();
+    pkg.cargo_metadata(true).probe("libevent_extra").unwrap();
+
+    if cfg!(feature = "openssl") {
+        pkg.cargo_metadata(true).probe("libevent_openssl").unwrap();
+    }
+
+    if cfg!(feature = "threading") {
+        pkg.cargo_metadata(true).probe("libevent_pthreads").unwrap();
+    }
+
+    let include_paths = lib.include_paths.drain(..).map(|path| {
+        let path_s = path.to_str().unwrap().to_string();
+        println!("cargo:include={}", &path_s);
+        path_s
+    }).collect();
+
+    Some(include_paths)
+}
+
+#[cfg(feature = "bundled")]
+fn find_libevent() -> Option<Vec<String>> {
+    use std::path::Path;
+    use std::process::Command;
+
+    if !Path::new("libevent/.git").exists() {
+        Command::new("git").args(&["submodule", "update", "--init"])
+            .status().expect("Running `git submodule init` failed.");
+    } else {
+        Command::new("git").args(&["submodule", "update", "--recursive"])
+            .status().expect("Running `git submodule update` failed.");
+    }
+
+    Some(vec![format!("{}/include", build_libevent("libevent").display())])
+}
+#[cfg(not(feature = "bundled"))]
+fn find_libevent() -> Option<Vec<String>> {
+    run_pkg_config()
+}
+
+fn main() {
+    println!("cargo:rerun-if-changed=build.rs");
```

Comment:
Specifying `cargo:rerun-if-changed=build.rs` is typically done to narrow the scope of change detection down to solely the build script, but other than that I believe it is redundant: 

> If the build script inherently does not need to re-run under any circumstance, then emitting cargo:rerun-if-changed=build.rs is a simple way to prevent it from being re-run. Cargo automatically handles whether or not the script itself needs to be recompiled, and of course the script will be re-run after it has been recompiled. Otherwise, specifying build.rs is redundant and unnecessary.

https://doc.rust-lang.org/cargo/reference/build-scripts.html#change-detection